Post transaction summary is logged for API users (RhBug:1855158)

m-blaha opened the pull request:
Move logging the post transaction summary from Cli to Base. After this change also packages changed via API (e.g. using Ansible) are logged into /var/log/dnf.log.

https://bugzilla.redhat.com/show_bug.cgi?id=1855158

pep8speaks commented:
Hello @m-blaha! Thanks for opening this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





* In the file [`dnf/cli/output.py`](https://github.com/rpm-software-management/dnf/blob/18435a16e26d23fcbc44f9cb3b9a4b21ee4e7431/dnf/cli/output.py):

> [Line 55:1](https://github.com/rpm-software-management/dnf/blob/18435a16e26d23fcbc44f9cb3b9a4b21ee4e7431/dnf/cli/output.py#L55): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1

j-mracek commented:
@rh-atomic-bot try

j-mracek commented:
@rh-atomic-bot retry

pkratoch commented:
The reproducer from the bug fails with:
```
Traceback (most recent call last):
  File "testdnfapi.py", line 13, in <module>
    base.do_transaction()
  File "/usr/lib/python3.8/site-packages/dnf/base.py", line 894, in do_transaction
    output = dnf.cli.output.Output(self, self.conf)
AttributeError: module 'dnf' has no attribute 'cli'
```

m-blaha commented:
Oh. I'm pretty sure it was working in the dbox environment but anway - you are right, the PR breaks dnf.

m-blaha commented:
I moved the post transaction output code from cli util.py. Now the code is called from two places and the behaviour is following:
- the last thing in Base.do_transaction() the output is written to the log file. For log file I changed the format and it now consists of lines `Action: NEVRA` for each package in the transaction. No columns, no multi-line output.
- the CLI output is called in BaseCli.do_transaction() and the format remained the same as before - packages are grouped by action and formatted into several columns (the number of columns depends on the terminal width).

pkratoch commented:
LGTM. Can you also add the changelog metadata to the last commit message? See: https://github.com/rpm-software-management/dnf#contribution

pkratoch commented:
@rh-atomic-bot try

m-blaha commented:
Yes, sure. I've added the changelog message.
